core/rawdb/freezer.go

```diff
@@ -226,7 +226,16 @@ func (f *Freezer) AncientSize(kind string) (uint64, error) {
 	defer f.writeLock.RUnlock()
 
 	if table := f.tables[kind]; table != nil {
-		return table.size()
+		// Get the current size of the table (including tail-deleted elements).
+		totalSize, _ := table.size()
+
+		// Get the number of tail-deleted elements.
+		tailDeletedSize := table.itemHidden.Load()
```

Comment:
The `size()` returns `size returns the total data size in the freezer table.` -- that is, a count of `bytes`. 
The `itemHidden`, however,  `is the number of items marked as deleted. -- a count of 'items'. 

So unfortunately you cannot just subtract the one from the other